File: js/structure-in-rte/ImplementLinkResolver.js
// DocSection: structure_in_rte_implement_link_resolver
// This is a global resolver for all links to articles. You can also define a resolver for a given query: https://github.com/Kentico/kontent-delivery-sdk-js/blob/master/DOCS.md#resolving-url-slugs-on-query-level

class Article extends ContentItem {
    constructor() {
        super({
            urlSlugResolver: (link, context) => {
                return { url: `/articles/${url_slug}` };
            }
        })
    }
}

// EndDocSection

File: ruby/structure-in-rte/ImplementLinkResolver.rb
# DocSection: structure_in_rte_implement_link_resolver
# Tip: Find more about Ruby SDKs at https://kontent.ai/learn/ruby
require 'delivery-sdk-ruby'

link_resolver = Kentico::Kontent::Delivery::Resolvers::ContentLinkResolver.new(lambda do |link|
        # Link is available
        return "/articles/#{link.url_slug}" if link.type.eql? 'article'
      end, lambda do |id|
        # Link is broken
        return "/notfound?id=#{id}"
      end)
# EndDocSection